webapp/src/Controller/BaseController.php

```diff
@@ -488,4 +491,67 @@ protected function streamResponse(callable $callback): StreamedResponse
         $response->setCallback($callback);
         return $response;
     }
+
+    protected function judgeRemaining(array $judgings): void
+    {
+        $inProgress = [];
+        $alreadyRequested = [];
+        $invalidJudgings = [];
+        $numRequested = 0;
+        foreach ($judgings as $judging) {
+            $judgingId = $judging->getJudgingid();
+            if ($judging->getResult() === null) {
+                $inProgress[] = $judgingId;
+            } elseif ($judging->getJudgeCompletely()) {
+                $alreadyRequested[] = $judgingId;
+            } elseif (!$judging->getValid()) {
+                $invalidJudgings[] = $judgingId;
+            } else {
+                $numRequested = $this->em->getConnection()->executeStatement(
+                    'UPDATE judgetask SET valid=1'
+                    . ' WHERE jobid=:jobid'
+                    . ' AND judgehostid IS NULL',
+                    [
+                        'jobid' => $judgingId,
+                    ]
+                );
+                $judging->setJudgeCompletely(true);
+                $this->em->flush();
+
+                $submission = $judging->getSubmission();
+
+                $queueTask = new QueueTask();
+                $queueTask->setJobId($judging->getJudgingid())
+                    ->setPriority(JudgeTask::PRIORITY_LOW)
+                    ->setTeam($submission->getTeam())
+                    ->setTeamPriority((int)$submission->getSubmittime())
+                    ->setStartTime(null);
+                $this->em->persist($queueTask);
+                $this->em->flush();
```

Comment:
We could do the flush after the loop. I think it’s faster to only flush once

Reply:
There were to flushed, both are now removed but I'm not sure if the 1st should actually be removed? And if we don't remove the first the amount of flushes is only halved (+1).

Reply:
One flush at the end should be enough I guess